sw/device/lib/base/freestanding/string.h

```diff
@@ -0,0 +1,26 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_BASE_FREESTANDING_STRING_H_
+#define OPENTITAN_SW_DEVICE_LIB_BASE_FREESTANDING_STRING_H_
+
+#include <stddef.h>
+
+/**
+ * @file
+ * @brief C library string handling (Freestanding)
+ *
+ * This header implements the string.h standard header, as required by C24 S4p7,
+ * and to the extent implemented by `sw/device/lib/base/memory.h`.
```

Comment:
I wonder if we should move these functions to here, but we could probably do that in follow up PRs.